### PR TITLE
Add pip_install_keeping_cached_packages query for Docker

### LINKED ISSUE #1067: Add pip_install_keeping_cached_packages query for Docker
### Platform
Docker

### Provider
Dockerfile

### Description
When installing packages with `pip`, the `--no-cache-dir` flag should be set to make Docker images smaller

### PR DESCRIPTION
**Pull request by mcarvalhox**
This query evaluates if the `--no-cache-dir` flag is set when installing packages with pip (to make Docker images smaller) . Closes #1067

### CODE REVIEW

File: assets/queries/dockerfile/disable_pip_cache/metadata.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "id": "disable_pip_cache",
+  "queryName": "Disable Pip Cache",
```
Comment (joaoReigota1) [inline assets/queries/dockerfile/disable_pip_cache/metadata.json:3]: The query name should reflect the vulnerability, not the fix

File: assets/queries/dockerfile/disable_pip_cache/query.rego [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+hasCacheFlag(values){
+	commands = split(values,"&&")
+    
+	some i
+    	instruction = trim_space(commands[i])
+      startswith(instruction, "pip")
+      contains(instruction, "install")
+      not contains(instruction, "--no-cache-dir")
```
Comment (joaoReigota1) [inline assets/queries/dockerfile/disable_pip_cache/query.rego:28]: The query is missing check for, "python -m pip install"

File: assets/queries/dockerfile/disable_pip_cache/test/positive_expected_result.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+[
+	{
+		"queryName": "Disable Pip Cache",
+		"severity": "MEDIUM",
+		"line": 1
```
Comment (joaoReigota1) [inline assets/queries/dockerfile/disable_pip_cache/test/positive_expected_result.json:5]: the bug with the docker parser is fixed should be pointing to line 2, not 1.
Merge with the master branch.

**Review by joaoReigota1 — CHANGES_REQUESTED** [review]

File: assets/queries/dockerfile/disable_pip_cache/test/positive_expected_result.json (hunk 3 above)
Comment (mcarvalhox) [reply assets/queries/dockerfile/disable_pip_cache/test/positive_expected_result.json:5]: Fixed in https://github.com/Checkmarx/kics/pull/1410

File: assets/queries/dockerfile/disable_pip_cache/query.rego (hunk 2 above)
Comment (mcarvalhox) [reply assets/queries/dockerfile/disable_pip_cache/query.rego:28]: Fixed in https://github.com/Checkmarx/kics/pull/1410

File: assets/queries/dockerfile/disable_pip_cache/metadata.json (hunk 1 above)
Comment (mcarvalhox) [reply assets/queries/dockerfile/disable_pip_cache/metadata.json:3]: Fixed in https://github.com/Checkmarx/kics/pull/1410

**Comment by ruigomescx** [comment]
Closed due to new PR #1410

**Pull request closed without merging**